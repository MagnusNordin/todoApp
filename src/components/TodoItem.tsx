import React, { FunctionComponent } from "react"
import styled from "styled-components"
import { useAppDispatch } from "../app/hooks"
import { completeTodo, removeTodo } from "../api/todoSlice"

import { Todo } from "../types/Todo"

type TodoItemProps = {
  item: Todo
}

const CompletedItem = styled.div`
  flex: 0 0 24px;
`
const ItemTitle = styled.div`
  text-align: left;
  flex: 1;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
`

const ItemWrapper = styled.section`
  display: flex;
  gap: 5px;
  padding-left: 5px;
  height: 40px;
`

const ItemUser = styled.div`
  margin-right: 10px;
`
export const TodoItem: FunctionComponent<TodoItemProps> = ({ item }) => {
  const dispatch = useAppDispatch()

  return (
    <ItemWrapper>
      {item.completed ? (
        <CompletedItem
          onClick={() =>
            dispatch(completeTodo({ id: item.id, completed: !item.completed }))
          }
        >
          &#9989;
        </CompletedItem>
      ) : (
        <CompletedItem
          onClick={() =>
            dispatch(completeTodo({ id: item.id, completed: !item.completed }))
          }
        >
          &#11093;
        </CompletedItem>
      )}
      <ItemTitle>{item.title}</ItemTitle>
      <ItemUser>{item.userId}</ItemUser>
      <div onClick={() => dispatch(removeTodo(item.id))}>&#x1F5D1;</div>
    </ItemWrapper>
  )
}
