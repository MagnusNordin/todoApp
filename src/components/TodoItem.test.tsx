import { render } from "@testing-library/react"
import { Provider } from "react-redux"
import { store } from "../app/store"
import { TodoItem } from "./TodoItem"

test("todo item", async () => {
  const element = { completed: false, id: 3, title: "tes", userId: 3 }
  const { asFragment } = render(
    <Provider store={store}>
      <TodoItem item={element} />
    </Provider>,
  )

  expect(asFragment).toMatchSnapshot()
})
