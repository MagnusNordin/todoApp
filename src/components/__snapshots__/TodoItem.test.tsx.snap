// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`todo item 1`] = `[Function]`;
